hpx/runtime/resource/partitioner.hpp

```diff
@@ -259,6 +259,9 @@ namespace hpx { namespace resource
         // Access all available NUMA domains
         HPX_EXPORT std::vector<numa_domain> const& numa_domains() const;
 
+        HPX_EXPORT std::size_t get_number_requested_threads();
+        HPX_EXPORT const hpx::threads::topology &get_topology() const;
```

Comment:
What's wrong with https://github.com/STEllAR-GROUP/hpx/blob/062f097c88fbe5de8c8a8e88dd42ee7cc18b203d/hpx/runtime/threads/topology.hpp#L426 or https://github.com/STEllAR-GROUP/hpx/blob/062f097c88fbe5de8c8a8e88dd42ee7cc18b203d/hpx/runtime.hpp#L152-L155?

Reply:
Also, I don't want to start a religious war about 'east const' and 'west const', but I would like to emphasize that all of HPX uses 'east const', thus it might be appropriate if you adjusted your style to this.

Reply:
At startup the user may have code that get's the RP, and also needs to get the topology to find out details about the node. The user shiouldn't need to know that the topology is available from the runtime (that hasn't been created yet). IMHO the topology should be available from the resource_partitioner as that's the place where the main management of the node takes place and development of our newer thread pool creation etc is taking place.
Getting it from the runtime ought to be deprecated in favour of the resource partitioner.

Reply:
By the way, there's one way to get it: https://github.com/STEllAR-GROUP/hpx/blob/062f097c88fbe5de8c8a8e88dd42ee7cc18b203d/examples/resource_partitioner/shared_priority_queue_scheduler.hpp#L1371-L1372, which as far as I can tell is equivalent to the one you've added? The difference seems to be that yours is in `partitioner` whereas the other one is in `detail::partitioner` which `hpx::resource::get_partitioner` returns.

Reply:
ok, then I will remove mine. I must have not found the other one when I was looking for it.

Reply:
OK. I've fixed the `var const &` east/west thingy and left the `get_topology` call in there - as discussed with @msimberg - if the user does
```
hpx::resource::partitioner rp(desc_cmdline, argc, argv);
```
where `rp` is a `resource::partitioner` - then it's a bit odd that they then have to call
```
auto &rp = resource::get_partitioner(); 
auto const& topo = rp.get_topology(); 
```
to get a detail partitioner and retrieve the topology from it. We probably should not be exposing `detail` level objects to the user.